src/utils/wallets.ts

```diff
@@ -32,16 +32,18 @@ export const WalletNames = {
 export const isWalletUnlocked = async (walletName: string): Promise<boolean> => {
   if (typeof window === 'undefined') return false
 
-  if (window.ethereum?.isConnected?.()) {
-    return true
-  }
-
   // Only MetaMask exposes a method to check if the wallet is unlocked
   if (walletName === WalletNames.METAMASK) {
-    return window.ethereum?._metamask?.isUnlocked?.() || false
+    return (await window.ethereum?._metamask?.isUnlocked?.()) || false
+  }
+
+  // WalletConnect v2
+  if (walletName === WalletNames.WALLET_CONNECT_V2) {
+    const wcSession = window.localStorage.getItem('wc@2:client:0.3//session')
```

Comment:
It looks like the `0.3` is [versioning](https://github.com/WalletConnect/walletconnect-monorepo/blob/06ca6bbf5aa26e869a4e1bfd83b5cca38eae7960/packages/core/src/controllers/store.ts#L74C1-L74C1).

You can call [`storageKey`](https://github.com/WalletConnect/walletconnect-monorepo/blob/06ca6bbf5aa26e869a4e1bfd83b5cca38eae7960/packages/core/src/controllers/store.ts#L73) (and maybe even [`get`](https://github.com/WalletConnect/walletconnect-monorepo/blob/06ca6bbf5aa26e869a4e1bfd83b5cca38eae7960/packages/core/src/controllers/store.ts#L143)) from WC [here](https://github.com/WalletConnect/walletconnect-monorepo/blob/v2.0/packages/sign-client/src/controllers/session.ts#L9) to avoid outdated session data.

Reply:
I'm aware of the trade-offs here and would sooner check local storage directly rather than call some internal WC methods. They can be made more robust by searching for a key named `wc@2:client:*//session` perhaps.

Reply:
> I'm aware of the trade-offs here and would sooner check local storage directly rather than call some internal WC method.

`storageKey` would return the key with which you could check the `localStorage` directly.

> This can be made more robust by searching for a key named `wc@2:client:*//session` perhaps.

I'd personally prefer this.